### ISSUE TITLE
[regression] adding a log device seems to require feature@allocation_classes

### ISSUE THREAD

**Issue by avg-I** [openzfs/zfs#13822]
Type | Version/Name
 --- | ---
OS Name	| FreeBSD
OS Version	| 13.1-STABLE
Architecture	| amd64
OpenZFS Version	| 2.1.5

This is a pool that was created a long time ago and went through many upgrades of ZFS (the software), however I've been conservative with enabling new features. The pool used to have a log device and I did not have any problems with it even after feature@allocation_classes was added (but never enabled on the pool). Recently I had to remove the log device. Afterwards I could not add it back:
```
# zpool add pond log ada0p4
cannot add to 'pond': pool must be upgraded to add these vdevs
```
The pool is a simple 2-way mirror.

In fact, I have another pool with a similar history where I never removed a log device and it still functions properly:
```
# zpool status rpool       
  pool: rpool
 state: ONLINE
status: Some supported and requested features are not enabled on the pool.
        The pool can still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
        the pool may no longer be accessible by software that does not support
        the features. See zpool-features(7) for details.
  scan: scrub repaired 0B in 00:09:04 with 0 errors on Mon Jul 25 03:50:11 2022
config:

        NAME        STATE     READ WRITE CKSUM
        rpool       ONLINE       0     0     0
          mirror-0  ONLINE       0     0     0
            ada0p6  ONLINE       0     0     0
            ada1p6  ONLINE       0     0     0
        logs
          mirror-1  ONLINE       0     0     0
            ada1p3  ONLINE       0     0     0
            ada0p3  ONLINE       0     0     0

errors: No known data errors
```
```
#zpool get feature@allocation_classes rpool
NAME   PROPERTY                    VALUE                       SOURCE
rpool  feature@allocation_classes  disabled                    local
```

Seems like maybe just an overzealous check in `zpool add` logic?

**Comment by avg-I** [comment]
More info about _pond_ pool:
```
zpool status pond
  pool: pond
 state: ONLINE
status: Some supported and requested features are not enabled on the pool.
        The pool can still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
        the pool may no longer be accessible by software that does not support
        the features. See zpool-features(7) for details.
  scan: scrub repaired 0B in 08:29:54 with 0 errors on Mon Jul  4 12:32:59 2022
config:

        NAME                              STATE     READ WRITE CKSUM
        pond                              ONLINE       0     0     0
          mirror-0                        ONLINE       0     0     0
            diskid/DISK-PN2A31PAJ8ZDDTp3  ONLINE       0     0     0
            diskid/DISK-MK0351YVKNNX3Ap3  ONLINE       0     0     0

errors: No known data errors
```
```
# zpool get all pond 
NAME  PROPERTY                       VALUE                          SOURCE
pond  size                           2.70T                          -
pond  capacity                       78%                            -
pond  altroot                        -                              default
pond  health                         ONLINE                         -
pond  guid                           9570370676287618392            -
pond  version                        -                              default
pond  bootfs                         -                              default
pond  delegation                     on                             default
pond  autoreplace                    off                            default
pond  cachefile                      -                              default
pond  failmode                       wait                           default
pond  listsnapshots                  off                            default
pond  autoexpand                     off                            default
pond  dedupratio                     1.00x                          -
pond  free                           591G                           -
pond  allocated                      2.13T                          -
pond  readonly                       off                            -
pond  ashift                         0                              default
pond  comment                        -                              default
pond  expandsize                     -                              -
pond  freeing                        0                              -
pond  fragmentation                  33%                            -
pond  leaked                         0                              -
pond  multihost                      off                            default
pond  checkpoint                     -                              -
pond  load_guid                      18406995395285268898           -
pond  autotrim                       off                            default
pond  compatibility                  off                            default
pond  feature@async_destroy          enabled                        local
pond  feature@empty_bpobj            active                         local
pond  feature@lz4_compress           active                         local
pond  feature@multi_vdev_crash_dump  enabled                        local
pond  feature@spacemap_histogram     active                         local
pond  feature@enabled_txg            active                         local
pond  feature@hole_birth             active                         local
pond  feature@extensible_dataset     active                         local
pond  feature@embedded_data          active                         local
pond  feature@bookmarks              enabled                        local
pond  feature@filesystem_limits      enabled                        local
pond  feature@large_blocks           enabled                        local
pond  feature@large_dnode            disabled                       local
pond  feature@sha512                 disabled                       local
pond  feature@skein                  disabled                       local
pond  feature@userobj_accounting     disabled                       local
pond  feature@encryption             disabled                       local
pond  feature@project_quota          disabled                       local
pond  feature@device_removal         disabled                       local
pond  feature@obsolete_counts        disabled                       local
pond  feature@zpool_checkpoint       disabled                       local
pond  feature@spacemap_v2            disabled                       local
pond  feature@allocation_classes     disabled                       local
pond  feature@resilver_defer         enabled                        local
pond  feature@bookmark_v2            disabled                       local
pond  feature@redaction_bookmarks    disabled                       local
pond  feature@redacted_datasets      disabled                       local
pond  feature@bookmark_written       disabled                       local
pond  feature@log_spacemap           disabled                       local
pond  feature@livelist               disabled                       local
pond  feature@device_rebuild         disabled                       local
pond  feature@zstd_compress          disabled                       local
pond  feature@draid                  disabled                       local
```

**Comment by Yneeb** [comment]
I believe that this is a duplicate of #8666 (which definitely has a title that is difficult to search for). Unfortunately, the only workarounds that I'm aware of are enabling the feature or temporarily using an older version of ZFS to add the log device.

**Comment by avg-I** [comment]
Thank you for pointing that out. Closing this one.
